inc/form_answer.class.php

```diff
@@ -782,19 +782,35 @@ public function acceptAnswers($datas) {
 
    public function generateTarget()
    {
-      global $CFG_GLPI;
-
-      $success = true;
-
       // Get all targets
-      $form = new PluginFormcreatorForm();
-      $form->getFromDB($this->fields['plugin_formcreator_forms_id']);
-      $target        = new PluginFormcreatorTarget();
-      $found_targets = $target->getTargetsForForm($form);
+      $target_class = new PluginFormcreatorTarget();
+      $found_targets = $target_class->find('plugin_formcreator_forms_id = ' . $this->fields['plugin_formcreator_forms_id']);
 
       $CFG_GLPI['plugin_formcreator_disable_hook_create_ticket'] = '1';
       // Generate targets
       foreach($found_targets as $target) {
+         $obj = new $target['itemtype'];
+         $obj->getFromDB($target['items_id']);
+         $_SESSION['items_id'] = $target['items_id'];
+         $condition = PluginFormcreatorExtension::generationTicketControl($target);
+         if ($condition) {
+            if (!$obj->save($this)) {
+               return false;
+            }
+            $generated = true;
+         }
+      }
+      unset($_SESSION['element_id']);
+      if ($generated) {
+         Session::addMessageAfterRedirect(__('The form has been successfully saved!', 'formcreator'), true, INFO);
+      } else {
+            Session::addMessageAfterRedirect(__('Une erreur s’est produite : votre demande n’a pas généré de ticket et ne sera pas traitée. Veuillez contacter votre correspondant de la DSI habituel ou créer un ticket d’anomalie sur l’application SOSDSI en précisant
```

Comment:
I think SOSDSI does not exists in any company using GLPI. This message is too specific.